contrib/gazebo/gazebo_sawyer_env.py

```diff
@@ -0,0 +1,499 @@
+        self._set_model_pose(
+            model_name="target",
+            new_pose=Pose(
+                position=Point(x=self.goal[0], y=self.goal[1], z=self.goal[
+                    2])),
+            reference_frame='world')
+        self._reset_sim()
+        obs = self._get_obs()
+        initial_observation = obs['observation']
+
+        if self._control_mode != 'position':
+            rospy.wait_for_service('/gazebo/pause_physics')
+            try:
+                pause = rospy.ServiceProxy('/gazebo/pause_physics',
+                                           std_srvs.srv.Empty)
+                pause()
+            except rospy.ServiceException as e:
+                print("/gazebo/pause_physics service call failed")
+
+        return initial_observation.copy()
+
+    def _reset_sim(self):
+        """
+        reset the simulation
+        :return:
+        """
+        print("Reseting the simulation...")
+
+        self._move_to_start_position()
+
+        # Randomize start position of object
+        if self.has_object:
+            object_xypos = self.initial_gripper_pos[:2]
+            print("Object xypos: ", object_xypos)
+            print("Initial Gripper Pos: ", self.initial_gripper_pos[:2])
+            while np.linalg.norm(
+                    object_xypos - self.initial_gripper_pos[:2]) < 0.1:
+                object_xypos = self.initial_gripper_pos[:
+                                                        2] + self.np_random_generator.uniform(
+                                                            -self.obj_range,
+                                                            self.obj_range,
+                                                            size=2)
+            self._set_model_pose(
+                model_name="block",
+                new_pose=Pose(
+                    position=Point(
+                        x=object_xypos[0],
+                        y=object_xypos[1],
+                        z=self._initial_qpos['object0'][2])),
+                reference_frame='world')
+
+    @property
+    def action_space(self):
+        """
+        Returns a Space object
+        """
+        return self._action_space
+
+    @property
+    def observation_space(self):
+        """
+        Returns a Space object
+        """
+        return self._observation_space
+
+    # Functions that gazebo env asks to implement
+    #-----------------------------------------------------------------------------------
+    def _env_setup(self, initial_qpos):
+        # Generate the world
+        # Load the model
+        GazeboEnv.load_gazebo_sdf_model(
+            model_name="cafe_table",
+            model_pose=Pose(
+                position=Point(initial_qpos['table0'][0], initial_qpos[
+                    'table0'][1], initial_qpos['table0'][2])),
+            model_path=osp.join(
+                rospkg.RosPack().get_path('sawyer_sim_examples'), "models",
```

Comment:
please use only " " strings or ' ' strings throughout a file. we prefer ' ' for everything but large blocks of text to be read by humans.